src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/BalanceSheetServiceImpl.java

```diff
@@ -72,66 +83,152 @@ public BalanceSheet getBalanceSheet(String transactionId, String companyAccounts
+    private String getCurrentPeriodHeading(CompanyProfileApi companyProfile, boolean isSameYear) {
+        NextAccountsApi nextAccountsApi = companyProfile.getAccounts().getNextAccounts();
+        LocalDate currentPeriodEndOn = convertDateTimeToLocalDate(nextAccountsApi.getPeriodEndOn());
+        LocalDate currentPeriodStartOn = convertDateTimeToLocalDate(nextAccountsApi.getPeriodStartOn());
 
-        boolean isSameYear = false;
+        return accountsDatesHelper.generateBalanceSheetHeading(currentPeriodStartOn, currentPeriodEndOn, isSameYear);
+    }
 
-        if (companyProfile.getAccounts().getLastAccounts() != null &&
-                companyProfile.getAccounts().getLastAccounts().getPeriodEndOn() != null) {
+    private String getPreviousPeriodHeading(CompanyProfileApi companyProfile, boolean isSameYear) {
+        LastAccountsApi lastAccountsApi = companyProfile.getAccounts().getLastAccounts();
+        if (isMultipleYearFiler(companyProfile)) {
+            LocalDate previousPeriodStartOn = convertDateTimeToLocalDate(lastAccountsApi.getPeriodStartOn());
+            LocalDate previousPeriodEndOn = convertDateTimeToLocalDate(lastAccountsApi.getPeriodEndOn());
 
-            LocalDate previousPeriodEndOn =
-                convertDateTimeToLocalDate(companyProfile.getAccounts().getLastAccounts().getPeriodEndOn());
-            isSameYear = accountsDatesHelper.isSameYear(previousPeriodEndOn, currentPeriodEndOn);
+            return accountsDatesHelper.generateBalanceSheetHeading(previousPeriodStartOn, previousPeriodEndOn, isSameYear);
         }
+        return null;
+    }
 
-        String currentPeriodHeading = accountsDatesHelper.generateBalanceSheetHeading(currentPeriodStartOn, currentPeriodEndOn, isSameYear);
+    private boolean isSameYear(CompanyProfileApi companyProfile) {
+        if (isMultipleYearFiler(companyProfile)) {
+            LastAccountsApi lastAccountsApi = companyProfile.getAccounts().getLastAccounts();
+            LocalDate previousPeriodEndOn = convertDateTimeToLocalDate(lastAccountsApi.getPeriodEndOn());
 
-        BalanceSheetHeadings balanceSheetHeadings = new BalanceSheetHeadings();
-        balanceSheetHeadings.setCurrentPeriodHeading(currentPeriodHeading);
+            NextAccountsApi nextAccountsApi = companyProfile.getAccounts().getNextAccounts();
+            LocalDate currentPeriodEndOn = convertDateTimeToLocalDate(nextAccountsApi.getPeriodEndOn());
 
-        return balanceSheetHeadings;
+            return accountsDatesHelper.isSameYear(previousPeriodEndOn, currentPeriodEndOn);
+        }
+        return false;
     }
 
-    private LocalDate convertDateTimeToLocalDate(DateTime dateTime) {
+    private CompanyProfileApi getCompanyProfile(String companyNumber) throws ServiceException {
+        return companyService.getCompanyProfile(companyNumber);
+    }
+
+    private String createUri(UriTemplate uri, String transactionId, String companyAccountsId) {
```

Comment:
Is this method required? Since we're returning a single method call it may read more simply to just call the uri `expand` method inline

Reply:
I'd add to that by saying we'll be using `expand()` in many other places, and having a custom convenience method just to call `toString()` on the result feels unnecessary and will end up deviating from what we do in other places, reducing overall consistency. There's also the risk that `createUri()` gets duplicated in other services implementations.

Reply:
I added it just because I was following the DRY principle, but if you feel it isn't needed then I will remove.